File: internal/actions/setup/setup_test.go
package setup

import (
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
)

func TestResolveNodeVersionTrims(t *testing.T) {
	v, err := ResolveNodeVersion(" 20 ")
	require.NoError(t, err)
	assert.Equal(t, "20", v)
}

func TestResolveNodeVersionErrorsWhenEmpty(t *testing.T) {
	_, err := ResolveNodeVersion("   ")
	assert.EqualError(t, err, "node-version is required")
}

func TestReport(t *testing.T) {
	assert.Equal(t, "▸ Setup\n  project       demo\n  node version  20", Report("demo", "20"))
}

func TestReportFallsBackForEmptyProject(t *testing.T) {
	assert.Contains(t, Report("", "20"), "project       (unknown project)")
}

File: actions/setup/scripts/setup.test.mjs
import { test } from "node:test";
import assert from "node:assert/strict";
import { resolveNodeVersion, report, renderOutputs } from "./setup.mjs";

test("resolveNodeVersion trims and returns the requested version", () => {
  assert.equal(resolveNodeVersion(" 20 "), "20");
  assert.equal(resolveNodeVersion("18"), "18");
});

test("resolveNodeVersion throws when empty or undefined", () => {
  assert.throws(() => resolveNodeVersion(""), /required/);
  assert.throws(() => resolveNodeVersion(undefined), /required/);
});

test("report shows the project and node version", () => {
  assert.equal(report("demo", "20"), "▸ Setup\n  project       demo\n  node version  20");
});

test("report falls back when the project name is empty", () => {
  assert.match(report("", "20"), /project +\(unknown project\)/);
  assert.match(report(undefined, "20"), /project +\(unknown project\)/);
});

test("renderOutputs formats key=value lines with a trailing newline", () => {
  assert.equal(renderOutputs({ node_version: "20" }), "node_version=20\n");
  assert.equal(renderOutputs({ a: "1", b: "2" }), "a=1\nb=2\n");
});
